### ISSUE TITLE
Secure desktops don't work

### ISSUE THREAD

**Issue by ghost** [Oliver2213/NVDARemote#1]
I have two laptops, both running Windows 7 HP X64.
Here's what I did on the target machine.
I set the addon to connect to a self-hosted server and saved the options.
I then saved the configuration to the system configuration from within NVDA's general settings dialog.
Then I got another computer and connected to the target computer using the same version of the addon.

Expected result: When entering a secure desktop, NVDA on the controlling computer should be able to control the target machine.

Actual result: Nothing happens when I enter a secure desktop. I can't press keys or hear anything. When I disconnect and reconnect, I can hear the beep as though it's connected, but nothing else can be done unless I use the target machine to get rid of the secure desktop first.

A workaround is to turn off the secure desktop in the registry, and a link to an article explaining that can be found here, method 2 is the one I used. https://support.microsoft.com/en-us/kb/975788

**Comment by Oliver2213** [comment]
Thanks for this. From what I can tell, things should be pretty much the same as far as secure desktop is concerned - it shouldn't matter if the server is automatically self-hosted or set up manually.

I'll definitely look at that code and see what the differences are. I'll try this myself, but can you try creating a self hosted server manually (disable the automatic server, and do it from the connect dialog), and see if things change?

**Comment by ghost** [comment]
It works after I do that. I copied the settings over to the system
config on the target computer, as well.

Actually after that I changed back to automatic mode and saved the
configuration. Then I copied the config over to the system config
again. It broke. Then I deleted remote.ini from the system config, and
it works now in automatic mode as long as I don't copy the remote.ini
file over to the system config.

**Comment by Oliver2213** [comment]
So... What your saying is:  
The automatic self-hosting server feature works, as long as you don't have remote.ini in the system configuration?

**Comment by ghost** [comment]
Yeah. Or if it's not set to automatically connect in the system config, it should work.

**Comment by Oliver2213** [comment]
Hmm... I think I know why that might be.
So it looks like when a connection is made, details about it are written to a temp file, and the secure desktop instance of the addon just connects to that.  
If the SD instance is trying to self-host it's own server (when the normal one already has one up), that could cause problems. I'll see what I can do to resolve this so that you can safely set the secure desktop instance to auto self-host and it'll just ignore that and connect to whatever the temp file says.
